lib/rules/array-callback-return.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * @fileoverview Rule to enforce return statements in callbacks of array's methods
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+var TARGET_NODE_TYPE = /^(?:Arrow)?FunctionExpression$/;
+var TARGET_METHODS = /^(?:every|fi(?:lter|nd(?:Index)?)|map|reduce(?:Right)?|so(?:me|rt))$/;
```

Comment:
Wondering if it's worth doing a performance test of some kind to see if it would be faster to use an array and indexOf, or an object with the methods as keys, to avoid the use of the regex engine.